src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,229 @@
+package cacophony.grammar.syntaxtree
+
+import cacophony.utils.Location
+
+sealed class Expression(val range: Pair<Location, Location>) { // everything in cacophony is an expression
+    class Variable(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    // should Type also inherit expression? if so, can we do for example "; Bool;" in our language?
+    class Type(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableDeclaration(
```

Comment:
shouldn't it consist of `let` keyword and variable identifier?

Reply:
I think we do not need to keep let, if, while etc as separate class - from parent type it is already known what to do with children expression, so after parsing I think keeping them is unnecessary (also, 'let' by itself is not a valid expression)
However Im not sure, so if you are I can change it or we can discuss it on labs

Reply:
I'm pretty sure there is no need to keep keywords in ast nodes. They are only useful when trying to parse an expression, once it is parsed there are redundant, all an `variable` declaration needs to know is the identifier of the variable and its type (and possibly value), there is no need to store `let`, `:` and `=`